lib/awful/placement.lua

```diff
@@ -1137,7 +1137,7 @@ function placement.resize_to_mouse(d, args)
     remove_border(d, args, ngeo)
 
     -- Now, correct the geometry by the given size_hints offset
-    if d.apply_size_hints then
+    if d.apply_size_hints and ngeo.width ~= 0 and ngeo.height ~= 0 then
         local w, h = d:apply_size_hints(
             ngeo.width,
```

Comment:
@psychon  what about:

    ngeo.width == 0 and 1 or ngeo.width,
    ngeo.height == 0 and 1 or ngeo.height,

This would still cause issues, but less than skipping the size hints? Fixing the floating code to stop at `1x1` would fix the other half of the problem. I have a PR which touches that code (moves code around to remove ~4 circular `require()`), I will add a commit for it and see if I can submit it soon.

Reply:
Makes me wonder if the C code handles this correctly. Let's say I have a window with an aspect ratio of 1:2 and I call `c:apply_size_hints(1, 1)`. I would expect the C code to return `0, 1` (or `1,0`?), which would then still cause the issue / bug that is described here...